polkadot/xcm/xcm-executor/src/traits/asset_exchange.rs

```diff
@@ -37,6 +37,13 @@ pub trait AssetExchange {
 		want: &Assets,
 		maximal: bool,
 	) -> Result<AssetsInHolding, AssetsInHolding>;
+
+	/// Handler for quoting the exchange price of two assets.
+	///
+	/// - `asset1` The first asset.
+	/// - `asset2` The second asset.
+	/// - `maximal`: If `true`, then all of `asset1` should be used.
+	fn quote_exchange_price(asset1: &Asset, asset2: &Asset, maximal: bool) -> Option<u128>;
```

Comment:
@gui1117 I agree that the lack of similarities between this function and `exchange_asset` make it confusing to use.

The relationship between the two is the following:
```rust
quote(give, want, minimal) = required_give -> exchange(required_give_amount, want, minimal) ✅ 
quote(give, want, maximal) = resulting_want -> exchange(give, resulting_want, maximal) ✅ 
```
Although in the case of `maximal`, the `want` parameter is only used as a minimum you want to get.
Given this relationship, I think it reasonable to change the parameters from simple assets to collections of them.
Does it make sense now?

Maybe the weirdest part is how `quote`'s return value changes so much in meaning depending on the `maximal` flag, but I'm sure that won't be an issue if we just document it properly.

Reply:
The current `quote_exchange_price` is OK to me, and fulfill the current usecase.

I was thinking about maybe in the future if some code wants reserve the necessary token get a successful `exchange_asset(_, (some DOT and KSM), USDC, false)` then the current API could be too limiting.

Like maybe user has 10DOT and 100KSM and would need to reserve all 10DOT and 50KSM to be able to exchange to N USDC.

But it can be fine, API can be improve when the need arise.

I'm good with the proposed API.